sklearn/ensemble/tests/test_voting.py

```diff
@@ -674,3 +680,103 @@ def test_get_features_names_out_classifier_error():
     )
     with pytest.raises(ValueError, match=msg):
         voting.get_feature_names_out()
+
+
+# Metadata Routing Tests
+# ======================
+
+
+@pytest.mark.parametrize(
+    "Estimator, Child",
+    [(VotingClassifier, ConsumingClassifier), (VotingRegressor, ConsumingRegressor)],
+)
+def test_routing_passed_metadata_not_supported(Estimator, Child):
+    """Test that the right error message is raised when metadata is passed while
+    not supported when `enable_metadata_routing=False`."""
+
+    X = np.array([[0, 1], [2, 2], [4, 6]])
+    y = [1, 2, 3]
+
+    with pytest.raises(
+        ValueError, match="is only supported if enable_metadata_routing=True"
+    ):
+        Estimator(["clf", Child()]).fit(X, y, sample_weight=[1, 1, 1], metadata="a")
+
+
+@pytest.mark.usefixtures("enable_slep006")
+@pytest.mark.parametrize(
+    "Estimator, Child",
+    [(VotingClassifier, ConsumingClassifier), (VotingRegressor, ConsumingRegressor)],
+)
+@pytest.mark.parametrize("prop", ["sample_weight", "metadata"])
+def test_metadata_routing_for_voting_estimators(Estimator, Child, prop):
+    """Test that metadata is routed correctly for Voting*."""
+    X = np.array([[0, 1], [2, 2], [4, 6]])
+    y = [1, 2, 3]
+    sample_weight, metadata = [1, 1, 1], "a"
+
+    pass_sample_weight = [True, False]
+    pass_metadata = [False, True]
+
+    est = Estimator(
+        [
+            (
+                "sub_est1",
+                Child(registry=_Registry()).set_fit_request(
+                    sample_weight=pass_sample_weight[0], metadata=pass_metadata[0]
+                ),
+            ),
+            (
+                "sub_est2",
+                Child(registry=_Registry()).set_fit_request(
+                    sample_weight=pass_sample_weight[1], metadata=pass_metadata[1]
+                ),
+            ),
+        ]
+    )
+
+    est.fit(X, y, **{prop: sample_weight if prop == "sample_weight" else metadata})
+
+    for estimator, boolean_sample_weight, boolean_metadata in zip(
+        est.estimators, pass_sample_weight, pass_metadata
+    ):
+        kwargs = {}
+        if prop == "sample_weight" and boolean_sample_weight is True:
+            kwargs = {prop: sample_weight}
+        if prop == "metadata" and boolean_metadata is True:
+            kwargs = {prop: metadata}
```

Comment:
I believe that having two props (`sample_weight` and `metadata`) here doesn't contribute a new test case and makes the test unnecessarily long.
Thus, I'd like to get rid of `props` and only use `sample_weight`. Is there something that speaks against it?

Reply:
This test can functionally stay the same and be simplified with this change:

```patch
diff --git a/sklearn/ensemble/tests/test_voting.py b/sklearn/ensemble/tests/test_voting.py
index 375703ea7b..d4d1cf0faa 100644
--- a/sklearn/ensemble/tests/test_voting.py
+++ b/sklearn/ensemble/tests/test_voting.py
@@ -717,35 +717,23 @@ def test_metadata_routing_for_voting_estimators(Estimator, Child, prop):
     y = [1, 2, 3]
     sample_weight, metadata = [1, 1, 1], "a"
 
-    pass_sample_weight = [True, False]
-    pass_metadata = [False, True]
-
-    est = Estimator(
-        [
-            (
-                "sub_est1",
-                Child(registry=_Registry()).set_fit_request(
-                    sample_weight=pass_sample_weight[0], metadata=pass_metadata[0]
-                ),
-            ),
-            (
-                "sub_est2",
-                Child(registry=_Registry()).set_fit_request(
-                    sample_weight=pass_sample_weight[1], metadata=pass_metadata[1]
-                ),
-            ),
-        ]
-    )
+    est = Estimator([
+        (
+            "sub_est1",
+            Child(registry=_Registry()).set_fit_request(**{prop: True}),
+        ),
+        (
+            "sub_est2",
+            Child(registry=_Registry()).set_fit_request(**{prop: True}),
+        ),
+    ])
 
     est.fit(X, y, **{prop: sample_weight if prop == "sample_weight" else metadata})
 
-    for estimator, boolean_sample_weight, boolean_metadata in zip(
-        est.estimators, pass_sample_weight, pass_metadata
-    ):
-        kwargs = {}
-        if prop == "sample_weight" and boolean_sample_weight is True:
+    for estimator in est.estimators:
+        if prop == "sample_weight":
             kwargs = {prop: sample_weight}
-        if prop == "metadata" and boolean_metadata is True:
+        else:
             kwargs = {prop: metadata}
         # access sub-estimator in (name, est) with estimator[1]
         registry = estimator[1].registry
```